escrow/memory_storage.go

```diff
@@ -110,13 +109,101 @@ func (storage *memoryStorage) Clear() (err error) {
 }
 
 func (storage *memoryStorage) StartTransaction(conditionKeys []string) (transaction Transaction, err error) {
-	return nil, fmt.Errorf("Not implemented")
+	conditionKeyValues := make([]KeyValueData, len(conditionKeys))
+	for i, key := range conditionKeys {
+		value, ok, err := storage.Get(key)
+		if err != nil {
+			return nil, err
+		} else if !ok {
+			conditionKeyValues[i] = KeyValueData{Key: key, Value: "", Present: false}
+		} else {
+			conditionKeyValues[i] = KeyValueData{Key: key, Value: value, Present: true}
+		}
+
+	}
+	transaction = &memoryStorageTransaction{ConditionKeys: conditionKeys, ConditionValues: conditionKeyValues}
+	return transaction, nil
 }
 
+func getValueDataForKey(key string, update []KeyValueData) (data KeyValueData, present bool) {
+	for _, data := range update {
+		if strings.Compare(data.Key, key) == 0 {
+			return data, true
+		}
+	}
+	return data, false
+}
 func (storage *memoryStorage) CompleteTransaction(transaction Transaction, update []KeyValueData) (ok bool, err error) {
-	return false, fmt.Errorf("Not implemented")
+	originalValues := transaction.(*memoryStorageTransaction).ConditionValues
+	for _, olddata := range originalValues {
+		if olddata.Present {
+			//make sure the current value is the same as the value last read
+			currentValue, _, _ := storage.Get(olddata.Key)
```

Comment:
It makes sense to check for `err` and `ok`, `err` should be `nil` and `ok` should be `true`

Reply:
fixed this